src/ol/interaction/interaction.js

```diff
@@ -180,6 +181,12 @@ ol.interaction.Interaction.zoom = function(view, resolution, opt_anchor, opt_dur
 ol.interaction.Interaction.zoomByDelta = function(view, delta, opt_anchor, opt_duration) {
   var currentResolution = view.getResolution();
   var resolution = view.constrainResolution(currentResolution, delta, 0);
+  var resolutions = view.getResolutions();
+
+  resolution = ol.math.clamp(
+      resolution,
```

Comment:
it's very strict type check. should change `resolution || 0` for `resolution: undefined`?

Reply:
No, only do the clamp when `resolution !== undefined`.